lib/pltr/v2/actions/lines.js

```diff
@@ -42,8 +46,55 @@ export function editLineColor(id, color) {
   return { type: EDIT_LINE_COLOR, id, color }
 }
 
-export function reorderLines(lines, bookId) {
-  return { type: REORDER_LINES, lines, bookId }
+export const reorderLines =
+  (droppedPosition, originalPosition, lines, bookId) => (dispatch, getState) => {
+    const isConflictWithPinned = lines.find((line) => {
+      if (line.position >= 0 && line.isPinned && line.position == droppedPosition) {
+        return line
+      }
+    })
+    const isPinnedPlotline = lines.find((line) => {
+      if (line.position >= 0 && line.isPinned && line.position == originalPosition) {
+        return line
+      }
+    })
+
+    const bothPinned = lines.find((line) => {
+      if (
+        line.isPinned &&
+        line.position >= 0 &&
+        line.position == droppedPosition &&
+        isPinnedPlotline &&
+        isPinnedPlotline.position >= 0 &&
+        isPinnedPlotline.position == originalPosition
+      ) {
+        return line
+      }
+    })
+
+    if ((!bothPinned && isPinnedPlotline) || (!bothPinned && isConflictWithPinned)) {
+      return false
+    }
+
+    const reorderedLines = reorderList(droppedPosition, originalPosition, lines)
+    return dispatch({ type: REORDER_LINES, lines: reorderedLines, bookId })
+  }
+
+export const togglePinPlotline = (line, lines, bookId) => (dispatch, getState) => {
```

Comment:
Can we narrow this down so that you only supply the line?
You can select both the lines and the bookId from the state.